[v1.0.0b] - Recipe import fails on image for maangchi.com

### First Check

- [X] This is not a feature request
- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.

### What is the issue you are experiencing?

Trying to import a recipe from https://www.maangchi.com/recipe/kkanpunggi fails with the following error. I can see id+json data in the page source, but apparently the image is in an unexpected format. 

```
[2023-04-24 14:25:00 +0200] [354] [ERROR] Exception in ASGI application
Traceback (most recent call last):
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/uvicorn/protocols/http/httptools_impl.py", line 436, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/fastapi/applications.py", line 276, in __call__
    await super().__call__(scope, receive, send)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/middleware/gzip.py", line 24, in __call__
    await responder(scope, receive, send)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/middleware/gzip.py", line 44, in __call__
    await self.app(scope, receive, self.send_with_gzip)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "/app/mealie/routes/_base/routers.py", line 35, in custom_route_handler
    response = await original_route_handler(request)
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/fastapi/routing.py", line 237, in app
    raw_response = await run_endpoint_function(
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/fastapi/routing.py", line 163, in run_endpoint_function
    return await dependant.call(**values)
  File "/app/mealie/routes/recipe/recipe_crud_routes.py", line 174, in parse_recipe_url
    recipe, extras = await create_from_url(req.url)
  File "/app/mealie/services/scraper/scraper.py", line 33, in create_from_url
    new_recipe, extras = await scraper.scrape(url)
  File "/app/mealie/services/scraper/recipe_scraper.py", line 30, in scrape
    result = await scraper.parse()
  File "/app/mealie/services/scraper/scraper_strategies.py", line 207, in parse
    return self.clean_scraper(scraped_data, self.url)
  File "/app/mealie/services/scraper/scraper_strategies.py", line 153, in clean_scraper
    image=try_get_default(None, "image", None, cleaner.clean_image),
  File "/app/mealie/services/scraper/scraper_strategies.py", line 122, in try_get_default
    value = clean_func(value)
  File "/app/mealie/services/scraper/cleaner.py", line 111, in clean_image
    raise TypeError(f"Unexpected type for image: {type(image)}, {image}")
TypeError: Unexpected type for image: <class 'list'>, [{'@id': 'https://i.ytimg.com/vi/VjneaJ0hmgs/maxresdefault.jpg?meta=og:image#sso/image-object'}]
```

### Deployment

Other

### Deployment Details

Running version 1.0.0b3 via the Home Assistant docker plugin.

--- a/mealie/services/scraper/cleaner.py
+++ b/mealie/services/scraper/cleaner.py
@@ -89,10 +89,10 @@ def clean_image(image: str | list | dict | None = None, default: str = "no image
     image attempts to parse the image field from a recipe and return a string. Currenty
 
     Supported Structures:
-        - `https://exmaple.com` - A string
-        - `{ "url": "https://exmaple.com" }` - A dictionary with a `url` key
-        - `["https://exmaple.com"]` - A list of strings
-        - `[{ "url": "https://exmaple.com" }]` - A list of dictionaries with a `url` key
+        - `https://example.com` - A string
+        - `{ "url": "https://example.com" }` - A dictionary with a `url` key
+        - `["https://example.com"]` - A list of strings
+        - `[{ "url": "https://example.com" }]` - A list of dictionaries with a `url` key
 
     Raises:
         TypeError: If the image field is not a supported type a TypeError is raised.
@@ -108,12 +108,12 @@ def clean_image(image: str | list | dict | None = None, default: str = "no image
             return [image]
         case [str(_), *_]:
             return [x for x in image if x]  # Only return non-null strings in list
-        case [{"@id": str(_)}, *_]:
-            return [x["@id"] for x in image]
         case [{"url": str(_)}, *_]:
             return [x["url"] for x in image]
         case {"url": str(image)}:
             return [image]
+        case [{"@id": str(_)}, *_]:
+            return [x["@id"] for x in image]
         case _:
             logger.exception(f"Unexpected type for image: {type(image)}, {image}")
             return [default]

--- a/mealie/services/scraper/cleaner.py
+++ b/mealie/services/scraper/cleaner.py
@@ -108,6 +108,8 @@ def clean_image(image: str | list | dict | None = None, default: str = "no image
             return [image]
         case [str(_), *_]:
             return [x for x in image if x]  # Only return non-null strings in list
+        case [{"@id": str(_)}, *_]:
+            return [x["@id"] for x in image]
         case [{"url": str(_)}, *_]:
             return [x["url"] for x in image]
         case {"url": str(image)}:

--- a/mealie/services/scraper/cleaner.py
+++ b/mealie/services/scraper/cleaner.py
@@ -115,7 +115,8 @@ def clean_image(image: str | list | dict | None = None, default: str = "no image
         case {"url": str(image)}:
             return [image]
         case _:
-            raise TypeError(f"Unexpected type for image: {type(image)}, {image}")
+            logger.exception(f"Unexpected type for image: {type(image)}, {image}")
+            return [default]
 
 
 def clean_instructions(steps_object: list | dict | str, default: list | None = None) -> list[dict]:

